mcstatus/motd/transformers.py

```diff
@@ -0,0 +1,303 @@
+        self.on_reset.append(f"</{tag_name}>")
+        return f"<{tag_name}>"
+
+
+class AnsiTransformer(PlainTransformer):
+
+    FORMATTING_TO_ANSI_TAGS = {
+        Formatting.BOLD: "1",
+        Formatting.STRIKETHROUGH: "9",
+        Formatting.ITALIC: "3",
+        Formatting.UNDERLINED: "4",
+        Formatting.OBFUSCATED: "5",
+    }
+    MINECRAFT_COLOR_TO_RGB = {
+        MinecraftColor.BLACK: (0, 0, 0),
+        MinecraftColor.DARK_BLUE: (0, 0, 170),
+        MinecraftColor.DARK_GREEN: (0, 170, 0),
+        MinecraftColor.DARK_AQUA: (0, 170, 170),
+        MinecraftColor.DARK_RED: (170, 0, 0),
+        MinecraftColor.DARK_PURPLE: (170, 0, 170),
+        MinecraftColor.GOLD: (255, 170, 0),
+        MinecraftColor.GRAY: (170, 170, 170),
+        MinecraftColor.DARK_GRAY: (85, 85, 85),
+        MinecraftColor.BLUE: (85, 85, 255),
+        MinecraftColor.GREEN: (85, 255, 85),
+        MinecraftColor.AQUA: (85, 255, 255),
+        MinecraftColor.RED: (255, 85, 85),
+        MinecraftColor.LIGHT_PURPLE: (255, 85, 255),
+        MinecraftColor.YELLOW: (255, 255, 85),
+        MinecraftColor.WHITE: (255, 255, 255),
+        MinecraftColor.MINECOIN_GOLD: (221, 214, 5),
+    }
+    RGB_TO_ANSI_8_BIT: dict[tuple[int, int, int], str] = {
+        (0, 0, 0): "0",
+        (0, 0, 170): "19",
+        (0, 170, 0): "34",
+        (0, 170, 170): "37",
+        (170, 0, 0): "124",
+        (170, 0, 170): "127",
+        (255, 170, 0): "214",
+        (170, 170, 170): "248",
+        (85, 85, 85): "240",
+        (85, 85, 255): "63",
+        (85, 255, 85): "83",
+        (85, 255, 255): "87",
+        (255, 85, 85): "203",
+        (255, 85, 255): "207",
+        (255, 255, 85): "227",
+        (255, 255, 255): "15",
+        (221, 214, 5): "184",
+    }
+    RGB_TO_ANSI_4_BIT: dict[tuple[int, int, int], str] = {
+        (0, 0, 0): "30",
+        (0, 0, 170): "34",
+        (0, 170, 0): "32",
+        (0, 170, 170): "36",
+        (170, 0, 0): "31",
+        (170, 0, 170): "35",
+        (255, 170, 0): "33",
+        (170, 170, 170): "37",
+        (85, 85, 85): "90",
+        (85, 85, 255): "94",
+        (85, 255, 85): "92",
+        (85, 255, 255): "96",
+        (255, 85, 85): "91",
+        (255, 85, 255): "95",
+        (255, 255, 85): "93",
+        (255, 255, 255): "97",
+        (221, 214, 5): "33",
+    }
+    RGB_TO_ANSI_3_BIT: dict[tuple[int, int, int], str] = {
+        (0, 0, 0): "30",
+        (0, 0, 170): "34",
+        (0, 170, 0): "32",
+        (0, 170, 170): "36",
+        (170, 0, 0): "31",
+        (170, 0, 170): "35",
+        (255, 170, 0): "1;33",
+        (170, 170, 170): "1;30",
+        (85, 85, 85): "1;30",
+        (85, 85, 255): "1;34",
+        (85, 255, 85): "1;32",
+        (85, 255, 255): "1;36",
+        (255, 85, 85): "1;31",
+        (255, 85, 255): "1;35",
+        (255, 255, 85): "33",
+        (255, 255, 255): "37",
+        (221, 214, 5): "1;33",
+    }
```

Comment:
Since color downscaling is already a planned feature, do we really need all of these dicts? Can't we just always go to full RGB, and simply use downscaling to then get the appropriate color. 

If these definitions are based off of the RGB ones, this should result in the exact same colors anyway. The only thing is that there will have to be some computation done for that downscaling, and I'm not sure how expensive that computation is. The current approach will certainly be faster, but if the downscaling isn't terribly slow, it's probably still a better option as it results in much cleaner code.